surefire-booter/src/main/java/org/apache/maven/surefire/booter/spi/EventChannelEncoder.java

```diff
@@ -128,7 +128,7 @@ public void onJvmExit()
     }
 
     @Override
-    public void systemProperties( Map<String, String> sysProps )
+    public void systemProperties( Map<String, String> sysProps, RunMode rm, Long testRunId )
```

Comment:
we already have field with the same name and type `runMode`
why not use filed from instance?

Reply:
field `runMode` sometime is used ... sometime is overridden by method params in this class
line: 389, 419

Reply:
The fields `runMode` will be deleted in `ForkinRunListener` and `EventChannelEncoder`.
This PR has only prepared changes in one method signature. The implementation of this abstract method comes right after.

Reply:
Your right, the `runMode` as a status should not be in 3 classes. It should be only in one, e.g. in the:
`JUnit4RunListener` which implements framework's listene `org.junit.runner.notification.RunListener`.

Reply:
ok, so it is next issue to clean up it.

Reply:
exactly